modules/api/pkg/resources/machine/machinedeployment.go

```diff
@@ -69,7 +69,7 @@ func Deployment(ctx context.Context, c *kubermaticv1.Cluster, nd *apiv1.NodeDepl
 	// Add Annotations to Machine Deployment
 	md.Annotations = nd.Annotations
 
-	osp := getOperatingSystemProfile(nd, dc)
+	osp := getOperatingSystemProfile(nd)
```

Comment:
```suggestion
	// Default OSP value if the OSP annotation on the MD is not preset.
	if osp := nd.Annotations[osmresources.MachineDeploymentOSPAnnotation]; osp == "" {
		osp := getOperatingSystemProfile(nd, dc)
```

Reply:
Something like this should be sufficient to fix this bug, needs testing though.

Reply:
> Default OSP value if the OSP annotation on the MD is not preset.

Default OSP value if the OSP annotation on the MD is not preset or present but empty. <-- something along these lines